### DISCUSSION TITLE
Allow small string to be used for ascii and multi byte without the utf logic

### DISCUSSION THREAD

**Discussion by xezon** [alandefreitas/small#51] [Ideas]
From what I have seen when stepping through code there is quite a bit computational overhead for the utf logic. I suspect this is not necessary when dealing with pure `ascii` and `multibyte` strings.

Perhaps make

`small::basic_string<char, N>` ascii
`small::basic_string<char8_t, N>` utf8
`small::basic_string<wchar_t, N>` multibyte
`small::basic_string<char16_t, N>` utf16

**Comment by alandefreitas** [comment]
Yes. A better alternative to all of that would be Boost.Text, but that didn't get into Boost because of Unicode copyright technicalities. 

The string classes in this library attempt to work with UTF natively but that causes more confusing than helps in most practical use cases.
